x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/detection_rules_client.ts

```diff
@@ -131,5 +148,14 @@ export const createDetectionRulesClient = ({
         });
       });
     },
+
+    async importRules(args: ImportRulesArgs): Promise<ImportRuleResponse[]> {
+      return withSecuritySpan('DetectionRulesClient.importRules', async () => {
+        return importRules({
+          ...args,
+          detectionRulesClient: this,
+        });
```

Comment:
Since this is an external method, we should either adapt it (making it more low-level and generic by removing response-related elements like status codes, bulk errors, etc.) or consider removing it from the rules management client entirely. This would help ensure a clearer separation of concerns and avoid unnecessary complexity in the client.

Reply:
That's fair. Do you have a preference? @banderror had originally suggested moving this to the client, since it already contained `importRule`, so my inclination is to the former (since this first-pass extraction maintained all of the route specifics, as you said), but the `importRules` function itself is quite specific to the route context so perhaps we should just leave it as a standalone function for now. Even if I managed to write a more agnostic `importRules`, adding back the status codes and proper errors would necessitate another pass over the data, so that's probably not ideal.

In the absence of further feedback, I will move this method from the client and instead directly call that function from the route.

Reply:
P.S. @xcrzx any thoughts on the `_with_source` suffix I've been using? I don't love it, and would appreciate alternate suggestions for naming here.

Reply:
I ended up separating the route specifics from the new client method in [this commit](https://github.com/elastic/kibana/pull/190198/commits/5c5e6371ef4fc1737068c0af131a1f212dbc60ca). @xcrzx can you please take a quick look and let me know if you have any feedback on that? I plan on adding/moving tests related to that refactor in the morning.

Reply:
Thanks for the refactoring - the separation looks much better now 👍

Reply:
Regarding `_with_source`. Do I understand correctly that the main difference between the two methods is that `importRuleWithSource` uses the `rule_source` value from the input rule, whereas `importRule` recalculates the `rule_source` value.

If so, wha about combining the two methods and reflecting the difference in method parameters. This would help us avoid duplication. For example:

```ts
importRule({
  ruleToImport,
  overwrite: {
    ruleSource: { ... },
    immutable: true,    
  }  
});
```

If `overwrite` parameters are provided, the method would use them; otherwise, it would recalculate the `rule_source` value. This approach keeps the logic in a single method while still allowing flexibility based on the input.

Reply:
c798d0e4431d36ddcb99d4eef7a1946e3a53eb8f and cfb709c1cb9268ec7bf567032655b5f6f86e8c09 should take care of this, which removes a lot of cruft from the branch. This is a nicer solution, thanks for the idea!